### PR TITLE
feat(cross-validation): Split time-related results into their own plots

### LINKED ISSUE #902: Split the CrossValidationReporter plot to have scores in one side, and time on the other side
### Is your feature request related to a problem? Please describe.

In the version to come (v0.5.0), the CrossValidationReporter has only one plot, with scores and time at the same place. It doesn't make much sense because 99.99% of the time, they will not have the same magnitude, therefore it's troublesome to see them at the same place.

### Describe the solution you'd like

I'd like to have two different plots. Keeping the current one: `CrossValidationReporter.plots.compare_scores`, but without the timings. And add a two new ones: `CrossValidationReporter.plots.timing` with the `fit_time` & `score_time`, and `CrossValidationReporter.plots.timing_normalized` with the `fit_time_per_data_point` & `score_time_per_data_point`

### Describe alternatives you've considered, if relevant

remove the time, but it's interesting so it would be a bummer to remove them.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by augustebaum**
Closes #902

Screen-record:

https://github.com/user-attachments/assets/00732aa4-a920-45b1-91b7-7d092f065827

- **fix(cross-validation): Remove "estimator" key even if "indices" is not present**
- **feat(cross-validation): Remove times from plot**
- **test: Test for CrossValidationItem's `plots` attribute**
- **feat: Group CrossValidationReporter plots into a single attribute**
- **feat: Add timing results plot function**
- **feat: Integrate timing results plot function to CrossValidationReporter**
- **feat: Add normalized timing results plot function**
- **feat: Integrate normalized timing results plot function to CrossValidationReporter**
- **refactor: Rename `plot_cross_validation` function**
- **refactor: Move cross-validation plots to their own module**

### CODE REVIEW

File: skore/src/skore/item/cross_validation_item.py [hunk 1]
```diff
@@ -152,7 +153,11 @@ def factory(cls, reporter):
...
         }
 
-        plot_bytes = plotly.io.to_json(plot, engine="json").encode("utf-8")
+        # FIXME: Maybe this logic belongs in CrossValidationPlots
+        plots_bytes = {
+            plot_name: plotly.io.to_json(plot, engine="json").encode("utf-8")
+            for plot_name, plot in dataclasses.asdict(plots).items()
+        }
 
         cv_info: dict[str, str] = {}
```
Comment (rouk1) [inline skore/src/skore/item/cross_validation_item.py:162]: FYI mypy warns here.

> By default the bodies of untyped functions are not checked, consider using --check-untyped-defsMypy(annotation-unchecked)

File: skore/src/skore/item/cross_validation_item.py [hunk 2]
```diff
@@ -152,7 +153,11 @@ def factory(cls, reporter):
             "hash": _hash_numpy(X_array),
         }
 
-        plot_bytes = plotly.io.to_json(plot, engine="json").encode("utf-8")
+        # FIXME: Maybe this logic belongs in CrossValidationPlots
```
Comment (rouk1) [inline skore/src/skore/item/cross_validation_item.py:156]: To me, this is ok as `CrossValidationPlots` is dataclass with no logic in it.

File: skore/src/skore/sklearn/cross_validation/plots/compare_scores_plot.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        _cv_results.pop(key, None)
+
+    df = pandas.DataFrame(_cv_results)
+
+    if "test_score" in df.columns:
+        df.insert(0, "test_score", df.pop("test_score"))
+
+    def linspace(lo, hi, num):
+        interval = (hi - lo) / (num - 1)
+        return [lo + k * interval for k in range(0, num)]
```
Comment (rouk1) [inline skore/src/skore/sklearn/cross_validation/plots/compare_scores_plot.py:48]: Maybe you can use [`numpy.linspace`](https://numpy.org/doc/2.1/reference/generated/numpy.linspace.html) here ?

File: skore/src/skore/sklearn/cross_validation/plots/compare_scores_plot.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+                    name=f"Average - 1 std. dev. {metric_name}",
+                    line=dict(dash="dot", color=bar_color),
+                    showlegend=False,
+                    mode="lines",
+                    **common_kwargs,
+                ),
+            ]
+        )
+
+    fig.update_xaxes(tickmode="linear", dtick=1, title_text="Split number")
```
Comment (rouk1) [inline skore/src/skore/sklearn/cross_validation/plots/compare_scores_plot.py:114]: IMO "Split number" is a bit confusing.

```suggestion
    fig.update_xaxes(tickmode="linear", dtick=1, title_text="Split index")
```

File: skore/src/skore/sklearn/cross_validation/plots/timing_normalized_plot.py [hunk 5]
```diff
@@ -125,13 +119,17 @@ def linspace(lo, hi, num):
...
                     showlegend=False,
                     mode="lines",
+                    hovertemplate="%{customdata}",
+                    customdata=[str(timedelta(seconds=avg_value + std_value))] * 10,
                     **common_kwargs,
                 ),
             ]
         )
 
     fig.update_xaxes(tickmode="linear", dtick=1, title_text="Split number")
```
Comment (rouk1) [inline skore/src/skore/sklearn/cross_validation/plots/timing_normalized_plot.py:129]: Same as above 
```suggestion
    fig.update_xaxes(tickmode="linear", dtick=1, title_text="Split index")
```

File: skore/src/skore/sklearn/cross_validation/plots/timing_plot.py [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+    df = pandas.DataFrame(_cv_results)
+
+    dict_labels = {
+        "fit_time": "fit_time (seconds)",
+        "score_time": "score_time (seconds)",
+    }
+
+    def linspace(lo, hi, num):
+        interval = (hi - lo) / (num - 1)
+        return [lo + k * interval for k in range(0, num)]
```
Comment (rouk1) [inline skore/src/skore/sklearn/cross_validation/plots/timing_plot.py:46]: Same as above could you use [`numpy.linspace`](https://numpy.org/doc/2.1/reference/generated/numpy.linspace.html) ?

File: skore/src/skore/sklearn/cross_validation/plots/timing_plot.py [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+                    showlegend=False,
+                    mode="lines",
+                    hovertemplate="%{customdata}",
+                    customdata=[str(timedelta(seconds=avg_value + std_value))] * 10,
+                    **common_kwargs,
+                ),
+            ]
+        )
+
+    fig.update_xaxes(tickmode="linear", dtick=1, title_text="Split number")
```
Comment (rouk1) [inline skore/src/skore/sklearn/cross_validation/plots/timing_plot.py:121]: Same as above.

```suggestion
    fig.update_xaxes(tickmode="linear", dtick=1, title_text="Split index")
```

**Review by rouk1 — COMMENTED** [review]
Nice work, just a few feedbacks.

File: skore/src/skore/item/cross_validation_item.py [hunk 8]
```diff
@@ -152,7 +153,11 @@ def factory(cls, reporter):
...
+        plots_bytes = {
+            plot_name: plotly.io.to_json(plot, engine="json").encode("utf-8")
+            for plot_name, plot in dataclasses.asdict(plots).items()
```
Comment (rouk1) [inline skore/src/skore/item/cross_validation_item.py:159]: Could the plot name be "humanized" ?
`timing_normalized` => _Normalized timings_ ?

File: skore/src/skore/sklearn/cross_validation/plots/compare_scores_plot.py (hunk 3 above)
Comment (augustebaum) [reply skore/src/skore/sklearn/cross_validation/plots/compare_scores_plot.py:48]: I did originally, but re-implementing it avoids bringing in numpy as a direct dependency for just 1 function

File: skore/src/skore/item/cross_validation_item.py (hunk 8 above)
Comment (augustebaum) [reply skore/src/skore/item/cross_validation_item.py:159]: You mean just in the front-end right? I mean, in their Python code user would still have to write
```python
reporter.plots.timing_normalized
```

File: skore/src/skore/sklearn/cross_validation/plots/compare_scores_plot.py (hunk 3 above)
Comment (augustebaum) [reply skore/src/skore/sklearn/cross_validation/plots/compare_scores_plot.py:48]: ...although I see that we import numpy in `cross_validation_item.py`...

File: skore/src/skore/item/cross_validation_item.py (hunk 1 above)
Comment (augustebaum) [reply skore/src/skore/item/cross_validation_item.py:161]: Added types

File: skore/src/skore/sklearn/cross_validation/plots/compare_scores_plot.py (hunk 4 above)
Comment (augustebaum) [reply skore/src/skore/sklearn/cross_validation/plots/compare_scores_plot.py]: Done

File: skore/src/skore/item/cross_validation_item.py (hunk 8 above)
Comment (rouk1) [reply skore/src/skore/item/cross_validation_item.py:158]: Yes only frontend side !

Comment (MarieS-WiMLDS) [reply skore/src/skore/item/cross_validation_item.py:158]: it completely makes sense! I like that it's easy from the human name to find how to call it in python. With that logic, maybe we could even switch from `compare_scores` to just `scores"?

Comment (augustebaum) [reply skore/src/skore/item/cross_validation_item.py:165]: Done (also renamed to `scores`)

File: skore/src/skore/sklearn/cross_validation/plots/compare_scores_plot.py (hunk 3 above)
Comment (augustebaum) [reply skore/src/skore/sklearn/cross_validation/plots/compare_scores_plot.py]: Done, added numpy to our direct deps

**Review by rouk1 — APPROVED** [review]
👍🏻

**Review by rouk1 — APPROVED** [review]

**Pull request merged**